GVFS/GVFS.NativeHooks.Common/common.posix.cpp

```diff
@@ -145,7 +145,11 @@ void DisableCRLFTranslationOnStdPipes()
 
 bool WriteToPipe(PIPE_HANDLE pipe, const char* message, size_t messageLength, /* out */ size_t* bytesWritten, /* out */ int* error)
 {
-    *bytesWritten = write(pipe, message, messageLength);
+    do
+    {
+        *bytesWritten = write(pipe, message, messageLength);
+    } while (*bytesWritten == -1 && errno == EINTR);
```

Comment:
Curious about this change?  Is this to block while the pipe is down?  Could this stop a process from completing?

Reply:
As best I could tell, we weren't properly handling `write` failures due to interrupts:

```
     [EINTR]            A signal interrupts the write before it could be completed.
```

See also: https://www.gnu.org/software/libc/manual/html_node/Interrupted-Primitives.html#Interrupted-Primitives

Reply:
As @jeffhostetler mentioned in a different comment:

> The mac version of GitService runs in byte/stream mode and not in dgram/message mode, so we can do the usual write-loop logic for short writes, rather than aborting if you want.

We are using stream mode here as well and shouldn't be aborting on short writes here either.